menu.py

```diff
@@ -100,39 +126,259 @@ def key_press(self, key: Key) -> Menu | None:
+                        f"{sensor_type.name.capitalize()} = {self._readings.get(sensor_type)}{unit}",
+                        col=2
+                    )
+
+    def update_sensor(self, sensor_type: SensorType):
+        """@brief update sensor sensor_type if currently shown on screen"""
+        with self._lock, self._display:
+            if self._current_menu is None:
+                if self.view is View.DUST and sensor_type in self.dust_view:
+                    self.display_view()
+                elif self.view is View.TEMP_PRES_HUMI and sensor_type in self.temp_view:
+                    self.display_view()
+
+
+class OnOffConfig(CallableMenuElement):
+    def __init__(self, display_str: str, config_value) -> None:
+        self.base_display_str = display_str
+        self.config_value = config_value
+        on_off = ConfigManager.get_config_value(config_value, display_config=True)
+        if on_off is None:
+            self.on_off = True
+            ConfigManager.update_config_values({config_value: str(int(True))}, display_config=True)
+        else:
+            self.on_off = bool(int(on_off))
+        super().__init__(f"{display_str}: {'ON' if self.on_off else 'OFF'}")
+
+    def call(self):
+        self.on_off = not self.on_off
+        ConfigManager.update_config_values({self.config_value: str(int(self.on_off))}, display_config=True)
+        self.display_str = f"{self.base_display_str}: {'ON' if self.on_off else 'OFF'}"
+
+
+class PoweroffMenu(CallableMenuElement):
+    def call(self):
+        signal.raise_signal(signal.SIGINT)
+        time.sleep(2.5)
+        os.system("sudo shutdown now -h")
+
+
+class RebootMenu(CallableMenuElement):
+    def call(self):
+        signal.raise_signal(signal.SIGINT)
+        time.sleep(2.5)
+        os.system("sudo reboot")
 
-    def key_press(self, key: Key) -> None:
-        if self.menu.key_press(key):
-            self.display.clear()
+
+class FreqencyChoice(Menu):
+    def __init__(self, display_str: str, display_config: bool, config_key: str, frequency_list: list[int]):
+        super().__init__("")
+        self.config_key = config_key
+        self.frequency_list = frequency_list
+        self.base_display_str = display_str
+        self.display_config = display_config
+        config_val = ConfigManager.get_config_value(config_key, display_config=display_config)
+        try:
+            # index of current freq
+            self.current_frequency = int(frequency_list.index(int(config_val))) if config_val else 0
+        except ValueError:
+            logging.warning("%s: Config value not present in list!", config_key)
+            self.current_frequency = 0
+        self.new_frequency = self.current_frequency
+
+        self._update_display_string()
+
+    def _update_display_string(self):
+        self.display_str = f"{self.base_display_str}: {self.frequency_list[self.current_frequency]}s"
+
+    def key_press(self, key: Key) -> Menu | None:
+        if key == Key.CANCEL:
```

Comment:
Hm, do you know if we have Python3 >= 3.10?
We would be able to do match case here, it would look match better imo.
Please consider.

Reply:
We could probably get any python we want but I don't think it would look better because if's are very simple and match/case would another indentation level.

Reply:
I have one more thing in mind, do we accept multiple buttons pressed at once?
If not, can we do `elif` instead to optimize?

Reply:
Each if ends with return. Even if it didn't this function is only called when button is pressed in frequency choice menu.
You can't push buttons fast enough to see any performance gain.
You probably wouldn't see any performance gain even if you could press buttons fast enough because redrawing part would take many times longer.
If you want to optimize then first run profiler and check what's most "cpu" hungry part.
I will probably correctly guess - drawing and access to files(that's why i made config cache and only load file if modified date changed since last time loaded)

Reply:
It ends with return, you're correct. However, I don't think it looks fancy that way. I mean, return is ok, but I still think else if should be used there even if there is no performance gain. Also, in the latter part:
```
if key == Key.DOWN:
else  #KEY.OK
```
I understand, that only 4 values are acceptable, else only addresses when it's not `Key.DOWN`. 
If we had if else if ... else it would cover !Key.UP, !DOWN, !CANCEL automatically (which is of course is not able to happen currently).
So I am still not convinced. Why not to use sth already provided by the language to improve readability? Else if automatically makes reviewer think that only one button handling at once is possible, he doesn't have to follow these returns.

Reply:
Ok, but don't complain because: `Unnecessary "elif" after "return", remove the leading "el" from "elif"pylint(no-else-return)`